lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -401,18 +399,59 @@ returns_boolean(Condition, Body) ->
 
 %% with
 
+translate_with(Meta, Args, S, false) ->
+  {Exprs, [{do, Do} | Opts]} = elixir_utils:split_last(Args),
+  {ElseClause, SE} = translate_with_else(Meta, Opts, S),
+  translate_with_do(Exprs, ?ann(Meta), Do, ElseClause, SE);
+translate_with(Meta, Args, S, true) ->
+  {Exprs, [{do, Do} | Opts]} = elixir_utils:split_last(Args),
+  case lists:any(fun is_clause_with_guard/1, Exprs) of
+    % we need to give up if has a guard
+    true -> translate_with(Meta, Args, S, false);
+    false -> translate_with_as_maybe(Exprs ++ [Do], ?ann(Meta), Opts, S)
+  end.
+
+% - `when` cannot be transpiled as maybe as far as I know (not supported)
```

Comment:
Gah, unfortunately not supporting guards is a deal breaker. :( Because it means we only compile some of the time and I would rather be consistent.